### PR TITLE
#252 : Adding Copy Novice

### PR DESCRIPTION
**Pull request by JonasAls89**

### CODE REVIEW

File: training/030_dtl/030_Novice.rst [hunk 1]
```diff
@@ -8,11 +8,59 @@ DTL: Novice
 "Copy"
 ~~~~~~
 
-Explain copy, based on ref 3.1.4 above
+Copy is one of the most fundamental DTL functions you will be using in Sesam when transforming your data. Copy might also be the most frequently used DTL function. The main reason for this is that it allows you to copy data by the use of wildcards, i.e. asterisk (*).
```
Comment (gabriellsesam) [inline training/030_dtl/030_Novice.rst:11]: `allows you to copy data by`
I suggest we should specify that we copy from the source to the target

File: training/030_dtl/030_Novice.rst [hunk 2]
```diff
@@ -8,11 +8,59 @@ DTL: Novice
...
 
-Wildcard namespace:\*
+In addition, ``["copy"]`` is convenient in that you can whitelist and blacklist data by providing arguments in the ``["copy"]`` function as follows:   
+
+``["copy", "*", "_*"]``
+
+The first argument in the above declaration accepts all values by the use of asterisk, whilst the second argument with the "_" prefix before the asterisk blacklists all values that has an "_" as part of their key property. For declarative purposes, this could be written as ``["copy", "<whitelist>", "<blacklist>"].`` The following example has been drafted to visually present this effect.
```
Comment (gabriellsesam) [inline training/030_dtl/030_Novice.rst:17]: `The first argument in the above declaration accepts all values by the use of asterisk`
->
`The first argument in the above declaration copies everything from your source to your output by the use of asterisk`

Comment (gabriellsesam) [inline training/030_dtl/030_Novice.rst:17]: `blacklists all values that has an "_" as part of their key property`
->
`blacklists all values whose keys start with "_"`

File: training/030_dtl/030_Novice.rst [hunk 3]
```diff
@@ -8,11 +8,59 @@ DTL: Novice
...
+
+Data entering a pipe from the system "sesam-training":
```
Comment (gabriellsesam) [inline training/030_dtl/030_Novice.rst:19]: Suggest to use the pipe name from the example, like:
`Data entering the pipe mssql-accounts`

File: training/030_dtl/030_Novice.rst [hunk 4]
```diff
@@ -8,11 +8,59 @@ DTL: Novice
...
+
+.. code-block:: json
+
+	{
+	  "mssql-accounts:country": "DK",
+	  "mssql-accounts:id": 40,
+	  "mssql-accounts:phone": "1-894-115-3398"
+	}
+
+As can be seen from the above produced data, the property with the key "_position" has been filtered by the ``["copy"]`` function and its arguments.
```
Comment (gabriellsesam) [inline training/030_dtl/030_Novice.rst:62]: maybe this is just subjective but
`from the above produced data`
->
`from the produced data above`

Comment (gabriellsesam) [inline training/030_dtl/030_Novice.rst:62]: `filtered by the ``["copy"]`` function and its arguments.`
->

filtered by the blacklist parameter `"_*"` in the ``["copy"]`` function.

**Review by gabriellsesam — CHANGES_REQUESTED** [review]
Very good example and good explanation, but I think it we can be more clear in some of the sentences.

I'd also love to see a sentence or two about copying all keys from a given namespace, like
`["copy", "golden-object:*"]`

File: training/030_dtl/030_Novice.rst (hunk 4 above)
Comment (JonasAls89) [reply training/030_dtl/030_Novice.rst:62]: Implemented everything aside from this comment.

**Review by gabriellsesam — APPROVED** [review]

**Review by ga-hegsvold — APPROVED** [review]

**Pull request merged**